cmd/integrationArtifactDeploy_test.go

```diff
@@ -39,23 +39,24 @@ func TestRunIntegrationArtifactDeploy(t *testing.T) {
 			Platform:               "cf",
 		}
 
-		httpClient := httpMockCpis{CPIFunction: "IntegrationArtifactDeploy", ResponseBody: ``, TestType: "Positive"}
+		httpClient := httpMockCpis{CPIFunction: "", ResponseBody: ``, TestType: "PostiveAndDeployIntegrationDesigntimeArtifactResBody"}
 
 		err := runIntegrationArtifactDeploy(&config, nil, &httpClient)
 
 		if assert.NoError(t, err) {
 
 			t.Run("check url", func(t *testing.T) {
-				assert.Equal(t, "https://demo/api/v1/DeployIntegrationDesigntimeArtifact?Id='flow1'&Version='1.0.1'", httpClient.URL)
+				assert.Equal(t, "https://demo/api/v1/IntegrationRuntimeArtifacts('flow1')", httpClient.URL)
 			})
 
 			t.Run("check method", func(t *testing.T) {
-				assert.Equal(t, "POST", httpClient.Method)
+				assert.Equal(t, "GET", httpClient.Method)
```

Comment:
added additional test case to support this

Reply:
I cannot find the new test (by searching for "`assert.Equal(t, "GET"`"), could you send me the commit hash.

Reply:
t.Run("Successfull Integration Flow Deploy Test", func(t *testing.T)  covers that

Reply:
But that is this test, no? So there is no additional test case?

Reply:
3 tests I added has complete code coverage. So I feel production code is well tested

Reply:
**t.Run("Trigger Failure for Integration Flow Deployment", func(t *testing.T)**  tests the deployment failure for POST request
			t.Run("check method", func(t *testing.T) {
				assert.Equal(t, "POST", httpClient.Method)
			})

Reply:
It would be easier for me if you sent a link to the line of code.

Reply:
[Found it here.](https://github.com/SAP/jenkins-library/pull/2628/files#diff-4b9118e22e14b055eceac87ba58e12e19349680ef2b5303f16acc13bbb5b9fffR81)

Reply:
> 3 tests I added has complete code coverage. So I feel production code is well tested

Code coverage is one thing, but I believe what is asserted has significance. @marcusholl could you comment on this? To my understanding the productive code will first trigger a POST request and then go into a loop, polling with GET requests. My opinion is that all http requests and their 'mocked' responses should be asserted. Also of interest might be to see that the polling works, so a check for the number of times the GET request was performed.

Reply:
Is it too complicated to implement these checks? If so I would suggest structuring the code differently. But maybe I'm just too picky, that is why another reviewers opinion would be helpful here. @radsoulbeard, @marcusholl

Reply:
line 80-82

Reply:
i have covered below parts in the Test. below 3 methods simulates all 4 http calls of production code

1. Assume deployment is successfull, get the deployment status. if status is started finish the test.

2. Assume deployment is successfull, get the deployment status. if status is failed, get the error details and finish the test

3. Assume deployment is failed, then check the error and check URL & Method.

Reply:
1. productive code will first trigger a POST request and then go into a loop
only when Artifact status is starting, then recursive call to same function would happen. this recursive behavior cannot be mocked. 
2. My opinion is that all http requests and their 'mocked' responses should be asserted. 
 we are mocking started state and failed state of artifact, but not starting state as that triggers recursive call.
3. Also of interest might be to see that the polling works, so a check for the number of times the GET request was performed.
since we are mocking http call which happens during recursive calls of function. this is not required i feel.